ts/Data/GoogleDataStore.ts

```diff
@@ -0,0 +1,268 @@
+/* *
+ *
+ *  Data module
+ *
+ *  (c) 2012-2020 Torstein Honsi
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import DataTable from './DataTable.js';
+import DataStore from './DataStore.js';
+import DataParser from './DataParser.js';
+import U from '../Core/Utilities.js';
+
+import AjaxMixin from '../Mixins/Ajax.js';
+const { ajax } = AjaxMixin;
+
+const {
+    fireEvent,
+    uniqueKey
+} = U;
+
+/** eslint-disable valid-jsdoc */
+
+/**
+ * @private
+ */
+
+class GoogleDataStore extends DataStore {
+    public constructor(
+        dataSet: DataTable,
+        options: GoogleSpreadsheetOptions
+    ) {
+        super(dataSet);
+
+        this.googleSpreadsheetKey = options.googleSpreadsheetKey;
+        this.worksheet = options.worksheet || 1;
+        this.startRow = options.startRow || 0;
+        this.endRow = options.endRow || Number.MAX_VALUE;
+        this.startColumn = options.startColumn || 0;
+        this.endColumn = options.endColumn || Number.MAX_VALUE;
+        this.enablePolling = options.enablePolling || false;
+        this.dataRefreshRate = options.dataRefreshRate || 2;
+        this.columns = [];
+    }
+
+    /* *
+    *
+    *  Properties
+    *
+    * */
+    public googleSpreadsheetKey: string;
+    public worksheet: number;
+    public startRow: number;
+    public endRow: number;
+    public startColumn: number;
+    public endColumn: number;
+    public enablePolling: boolean;
+    public dataRefreshRate: number;
+    public columns: Array<Array<Highcharts.DataValueType>>;
+
+    private dataParser = new DataParser();
+
+    /* *
+    *
+    *  Functions
+    *
+    * */
+    private getSheetColumns(json: Highcharts.JSONType): Array<Array<Highcharts.DataValueType>> {
```

Comment:
We need to integrated the types somewhere, so that we do not use the global Highcharts namespace.